packages/cdk-routing-split/test/cases/integration/get-origin.test.ts

```diff
@@ -125,4 +125,60 @@ describe('When an entity', () => {
         };
         expect(response).toStrictEqual(success);
     });
+
+    it('calls getOrigin with a valid url with different priorities', async () => {
+        const domainRequest = {
+            key: UrlSegment.Domain,
+            value: domainData.domain,
+            priority: 1,
+            origin: given.a_simple_url(),
+        } as AddRequest;
+        await when.we_invoke_add(domainRequest);
+
+        const queryRequest = {
+            key: UrlSegment.QueryStringParam,
+            value: domainData.queryStrings[0],
+            priority: 2,
+            origin: given.a_simple_url(),
+        } as AddRequest;
+        await when.we_invoke_add(queryRequest);
+
+        const pathRequest = {
+            key: UrlSegment.FirstPathSegment,
+            value: domainData.pathname,
+            priority: 100,
+            origin: given.a_simple_url(),
+        } as AddRequest;
+        await when.we_invoke_add(pathRequest);
+
+        const subdomainRequest = {
+            key: UrlSegment.Subdomain,
+            value: domainData.subdomain,
+            priority: 3,
+            origin: given.a_simple_url(),
+        } as AddRequest;
+        await when.we_invoke_add(subdomainRequest);
+
+        const origin = pathRequest.origin;
```

Comment:
This should be domain request. The lowest number has the highest priority, so the domainRequest's origin is what should be expected.